packages/flutter/lib/src/widgets/modal_button_sheet.dart

```diff
@@ -0,0 +1,652 @@
+import 'dart:async';
+import 'dart:ui';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/widgets.dart';
+
+const Curve _decelerateEasing = Cubic(0.0, 0.0, 0.2, 1.0);
```

Comment:
before reading this class, my first impression is that I'm tempted to think that we can just somewhat follow the https://github.com/flutter/flutter/blob/354e2a57b1ea20cda88ae1c24b5283dced54fde2/packages/flutter/lib/src/material/page_transitions_theme.dart#L561 example and not push something so visually opinionated in terms of implementation (such as curves, velocities etc) into the widgets layer. Having some general semantics of normal vs fullscreen vs modal sheet is ok but there's a lot of implementation details here. 

i.e. I have trouble imagining a case where someone wants to base a transition off of this one, not use material or cupertino and find these existing specifications in terms of velocity, duration etc useful.

Reply:
Yeah, that was my approach from the beginning, but a lot of people have been asking for this level of customization.  The package has them default from the beginning and has been adding these params durations, curves by demand.

If `modal_bottom_sheet.dart` is merged with `modal_bottom_sheet_route.dart`, `ModalBottomSheet` class could be private